src/shims/unix/solarish/foreign_items.rs

```diff
@@ -18,6 +19,30 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     ) -> InterpResult<'tcx, EmulateItemResult> {
         let this = self.eval_context_mut();
         match link_name.as_str() {
+            // Allocation
+            "memalign" => {
```

Comment:
With https://github.com/rust-lang/rust/pull/124798, `memalign` will not be called on Solarish any more.

You can still add it here if you want, but then you should also add a test that calls this function (directly via `libc`).

Reply:
This comment is not resolved yet: you're adding a new shim that's entirely untested. That's not going to fly.